src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -326,10 +330,80 @@ export const FieldToInteractionMap: {
 } = {
   name: ANT_INTERACTION_TYPES.SET_NAME,
   ticker: ANT_INTERACTION_TYPES.SET_TICKER,
+  targetID: ANT_INTERACTION_TYPES.SET_TARGET_ID,
   // TODO: add other interactions
 };
 
 export function getInteractionTypeFromField(field: string) {
   // TODO: add contract specification and more interaction fields
   return FieldToInteractionMap[field];
 }
+
+export function mapTransactionDataKeyToPayload(
+  contractType: ContractType,
+  interactionType: AntInteraction | RegistryInteraction,
+  data: string[],
+): TransactionData | undefined {
+  const payload: any = {};
+
+  if (isRegistryInteraction(interactionType)) {
+    if (
+      !isInteractionCompatible({
+        contractType,
+        interactionType,
+        functionName:
+          TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType]
+            .functionName,
+      })
+    ) {
+      throw new Error(`Interaction is not compatible`);
+    }
+    payload.functionName =
+      TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+        interactionType
+      ].functionName;
+    TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][
+      interactionType
+    ].keys.forEach((key, index) => {
+      if (!data[index]) {
+        // if missing transaction data from the url, throw an error
+        throw new Error(
+          `Missing key (${key}) from transaction data in the url. This may be due to the order of the data, the current order is [${data}], the correct order is [${
+            TRANSACTION_DATA_KEYS[CONTRACT_TYPES.REGISTRY][interactionType].keys
+          }]`,
+        );
+      }
+      payload[key] = data[index];
+    });
```

Comment:
i was hoping to see _this_ in it's own function that just takes the `CONTRACT_TYPE` AND `interactionType` to return the payload with all the keys. it's the exact same logic below 😢

Reply:
due to the funky nature of the TRANSACTION_DATA_KEYS object, this is necesary to duplicated for every contract instance. Unless you know of a better way.

Reply:
this makes me sad. i think there is opportunity to improve this so please add a `TODO` comment here. I'd imagine with some of @matibat 's type changes, we can more effectively implement this util.

Reply:
added todo

Reply:
added todo